### ISSUE TITLE
Problem with titleFormatter in ios 7.1

### ISSUE THREAD

**Issue by ostatnicky** [HeshamMegid/HMSegmentedControl#104]
It's the similar problem as issue [#92](https://github.com/HeshamMegid/HMSegmentedControl/issues/92) but for titleFormatter.

I have this code:

```
UIColor *greenColor = [UIColor greenColor];
UIColor *blueColor = [UIColor blueColor];
UIColor *backgroundColor = [UIColor grayColor];

NSArray *items = [NSArray arrayWithObjects:[NSLocalizedString(@"First", @"") uppercaseString], [NSLocalizedString(@"Second", @"") uppercaseString], nil];

HMSegmentedControl *segmentedControl = [[HMSegmentedControl alloc] initWithSectionTitles:items];
segmentedControl.frame = CGRectMake(0, 0, viewWidth, 38);
[segmentedControl addTarget:self action:@selector(didChangeValue:) forControlEvents:UIControlEventValueChanged];
[segmentedControl setTitleFormatter:^NSAttributedString *(HMSegmentedControl *segmentedControl, NSString *title, NSUInteger index, BOOL selected) {
    UIColor *textColor = (selected) ? greenColor : blueColor;

    NSAttributedString *attString = [[NSAttributedString alloc] initWithString:title attributes:@{NSForegroundColorAttributeName : textColor, NSFontAttributeName: [UIFont systemFontOfSize:17.0]}];
    return attString;
}];
segmentedControl.selectionIndicatorColor = greenColor;
segmentedControl.selectionIndicatorLocation = HMSegmentedControlSelectionIndicatorLocationDown;
segmentedControl.selectionIndicatorHeight = 3;
segmentedControl.selectionStyle = HMSegmentedControlSelectionStyleBox;
segmentedControl.selectionIndicatorBoxOpacity = 0;
segmentedControl.segmentEdgeInset = UIEdgeInsetsMake(0, 20, 0, 20);
segmentedControl.backgroundColor = backgroundColor;
[self.view addSubview:segmentedControl];
```

Log is saying:

```
HMSegmentedControlExample[6675] <Error>: CGContextSetFillColorWithColor: invalid context 0x0. This is a serious error. This application, or a library it uses, is using an invalid context  and is thereby contributing to an overall degradation of system stability and reliability. This notice is a courtesy: please fix this problem. It will become a fatal error in an upcoming update.
```

I tried it on 1.5.1.

**Comment by yaoandw** [comment]
in HMSegmentedControl.m
try replace
````
 titleLayer.string = [self attributedTitleAtIndex:idx];
````
with
````
 if ([[[UIDevice currentDevice] systemVersion] floatValue] >= 8.0) {
                titleLayer.string = [self attributedTitleAtIndex:idx];
            }else{
                NSAttributedString *str = [self attributedTitleAtIndex:idx];
                titleLayer.string = str.string;
                [str enumerateAttributesInRange:NSMakeRange(0, str.length) options:(0) usingBlock:^(NSDictionary *attrs, NSRange range, BOOL *stop) {
                    UIFont *font = attrs[NSFontAttributeName];
                    UIColor *color = attrs[NSForegroundColorAttributeName];
                    [titleLayer setFont:(__bridge CFTypeRef)(font.fontName)];
                    [titleLayer setFontSize:font.pointSize];
                    [titleLayer setForegroundColor:color.CGColor];
                }];
            }
````

**Comment by pranavss11** [comment]
Any updates on this? It would be nice to have a pod update for this @HeshamMegid

**Comment by HeshamMegid** [comment]
@pranavss11 I haven't had the time to work on a fix yet and test it on iOS 7, but feel free to  send a pull request and I'll merge it.

The correct fix should be similar to d8e6e4a14024309b6a9c68286d60cf660e4d342b.

**Comment by HeshamMegid** [comment]
This should be fixed in the latest release. Feel free to reopen if it's not.
